eventhubs/client/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClient.java

```diff
@@ -0,0 +1,292 @@
+ * Event Hub.
+ */
+public class EventHubClient implements Closeable {
+    /**
+     * The name of the default consumer group in the Event Hubs service.
+     */
+    public static final String DEFAULT_CONSUMER_GROUP_NAME = "$Default";
+
+    private static final String RECEIVER_ENTITY_PATH_FORMAT = "%s/ConsumerGroups/%s/Partitions/%s";
+    private static final String SENDER_ENTITY_PATH_FORMAT = "%s/Partitions/%s";
+
+    private final String connectionId;
+    private final Mono<EventHubConnection> connectionMono;
+    private final AtomicBoolean hasConnection = new AtomicBoolean(false);
+    private final ConnectionOptions connectionOptions;
+    private final String eventHubPath;
+    private final EventHubProducerOptions defaultProducerOptions;
+    private final EventHubConsumerOptions defaultConsumerOptions;
+
+    EventHubClient(ConnectionOptions connectionOptions, ReactorProvider provider, ReactorHandlerProvider handlerProvider) {
+        Objects.requireNonNull(connectionOptions);
+        Objects.requireNonNull(provider);
+        Objects.requireNonNull(handlerProvider);
+
+        this.connectionOptions = connectionOptions;
+        this.eventHubPath = connectionOptions.eventHubPath();
+        this.connectionId = StringUtil.getRandomString("MF");
+        this.connectionMono = Mono.fromCallable(() -> {
+            return (EventHubConnection) new ReactorConnection(connectionId, connectionOptions, provider, handlerProvider, new ResponseMapper());
+        }).doOnSubscribe(c -> hasConnection.set(true))
+            .cache();
+
+        this.defaultProducerOptions = new EventHubProducerOptions()
+            .retry(connectionOptions.retryPolicy())
+            .timeout(connectionOptions.timeout());
+        this.defaultConsumerOptions = new EventHubConsumerOptions()
+            .retry(connectionOptions.retryPolicy())
+            .scheduler(connectionOptions.scheduler());
+    }
+
+    /**
+     * Creates a builder that can configure options for the {@link EventHubClient} before creating an instance of it.
+     *
+     * @return A new {@link EventHubClientBuilder} to create an EventHubClient from.
+     */
+    public static EventHubClientBuilder builder() {
+        return new EventHubClientBuilder();
+    }
+
+    /**
+     * Retrieves information about an Event Hub, including the number of partitions present and their identifiers.
+     *
+     * @return The set of information for the Event Hub that this client is associated with.
+     */
+    public Mono<EventHubProperties> getProperties() {
+        return connectionMono.flatMap(connection -> connection.getManagementNode().flatMap(EventHubManagementNode::getEventHubProperties));
+    }
+
+    /**
+     * Retrieves the set of identifiers for the partitions of an Event Hub.
+     *
+     * @return The set of identifiers for the partitions of an Event Hub.
+     */
+    public Mono<String[]> getPartitionIds() {
```

Comment:
I wonder if this is more convenient as a Flux<String> - then people can more easily convert it to a stream or iterable.

Reply:
In the spirit of using reactive programming, It would probably align with that paradigm better.